scripts/split-e2e-tests-v2.ts

```diff
@@ -0,0 +1,314 @@
+        'delete_pkg',
+        'diagnose_pkg',
+        'env-2_pkg',
+        'export_pkg',
+        'function_1_pkg',
+        'function_3a_pkg',
+        'function_3b_pkg',
+        'function_4_pkg',
+        'function_6_pkg',
+        'function_7_pkg',
+        'function_8_pkg',
+        'geo-add-e_pkg',
+        'geo-add-f_pkg',
+        'geo-remove-2_pkg',
+        'geo-remove-3_pkg',
+        'geo-update-1_pkg',
+        'geo-update-2_pkg',
+        'git-clone-attach_pkg',
+        'hooks-a_pkg',
+        'import_auth_1a_pkg',
+        'import_auth_1b_pkg',
+        'import_auth_2a_pkg',
+        'import_auth_2b_pkg',
+        'import_auth_3_pkg',
+        'import_dynamodb_2a_pkg',
+        'import_dynamodb_2c_pkg',
+        'import_s3_2a_pkg',
+        'import_s3_2c_pkg',
+        'layer-2_pkg',
+        'mock-api_pkg',
+        'model-migration_pkg',
+        'notifications-analytics-compatibility-in-app-1_pkg',
+        'notifications-analytics-compatibility-sms-1_pkg',
+        'notifications-analytics-compatibility-sms-2_pkg',
+        'notifications-in-app-messaging-env-1_pkg',
+        'notifications-in-app-messaging-env-2_pkg',
+        'notifications-lifecycle_pkg',
+        'notifications-sms_pkg',
+        'notifications-sms-pull_pkg',
+        'pull_pkg',
+        'schema-iterative-rollback-1_pkg',
+        'schema-iterative-rollback-2_pkg',
+        'searchable-migration_pkg',
+        'storage-5_pkg',
+        'studio-modelgen_pkg',
+        'uibuilder_pkg',
+    ],
+}
+const MAX_WORKERS = 3;
+type OS_TYPE = 'w' | 'l';
+type CandidateJob = {
+    region: string,
+    os: OS_TYPE,
+    executor: string,
+    tests: string[],
+    useParentAccount: boolean,
+    // intentially leaving this here - accounts are randomly assigned to jobs 
+    // by a via local_publish_helpers.sh script
+    // account: string, 
+}
+type Job = CandidateJob & {
+    name: string;
+    requiredJobs: string[];
+}
+
+const createRandomJob = (os: OS_TYPE) : CandidateJob => {
+    const region = regions[Math.floor(Math.random() * regions.length)];
+    return {
+        region,
+        os,
+        executor: os === 'l' ? 'l_large' : 'w_medium',
+        tests: [],
+        useParentAccount: false,
+    }
+}
+
+
+
+export const splitTestsV2 = function splitTests(
+    config: Readonly<CircleCIConfig>,
+    counts: { w: number, l: number},
+    baseJobName: string,
+    workflowName: string,
+    jobRootDir: string,
+    isMigration: boolean,
+    pickTests: ((testSuites: string[]) => string[]) | undefined,
+  ): CircleCIConfig {
+    
+
+    const output: CircleCIConfig = { ...config };
+    const baseJobs = { ...config.jobs };
+    const baseJob = baseJobs[baseJobName];
+    
+    let testSuites = getTestFiles(jobRootDir);
+    if(pickTests && typeof pickTests === 'function'){
+        testSuites = pickTests(testSuites);
+    }
+    if(testSuites.length === 0){
+        return output;
+    }
+    const testFileRunTimes = loadTestTimings().timingData;
```

Comment:
this is important because we want similar timed jobs to be together
<img width="1550" alt="image" src="https://user-images.githubusercontent.com/110861985/210122538-7abad221-0617-4665-b2b9-a8b9ba8f0773.png">